### PR TITLE
Discovery: Add TAG sync object to Discovery config

### PR DESCRIPTION
**Pull request by tigrato**
This PR adds the ability to configure TAG AWS poll in discovery config.

Part of https://github.com/gravitational/access-graph/issues/458

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6536,3 +6536,21 @@ message OktaOptions {
...
+  // AccountID is the AWS account ID to sync.
+  string AccountID = 1 [(gogoproto.jsontag) = "account_id,omitempty"];
+  // Regions are AWS regions to import resources from.
+  repeated string Regions = 2 [(gogoproto.jsontag) = "regions,omitempty"];
+  // AssumeRoleARN is the AWS role to assume for database discovery.
+  string AssumeRoleARN = 3 [(gogoproto.jsontag) = "assume_role_arn,omitempty"];
+  // ExternalID is the AWS external ID to use when assuming a role for
+  // database discovery in an external AWS account.
+  string ExternalID = 4 [(gogoproto.jsontag) = "external_id,omitempty"];
+}
```
[the remark this answers is not in the document]
Comment (tigrato) [reply api/proto/teleport/legacy/types/types.proto:6556]: All the other matchers are defined here. I am not against moving them to other packages but will be weird having all except one here.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -6536,3 +6536,21 @@ message OktaOptions {
     (gogoproto.casttype) = "Duration"
   ];
 }
+
+// AccessGraphSync is a configuration for Access Graph service.
+message AccessGraphSync {
+  repeated AccessGraphAWSSync AWS = 1 [(gogoproto.jsontag) = "types,omitempty"];
```
Comment (jakule) [inline api/proto/teleport/legacy/types/types.proto:6542]: nit: missing "ProtoDoc"

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakule) [reply api/proto/teleport/legacy/types/types.proto:6556]: I agree with Tiago on this. I'm not a fan of GoGo, but this PR doesn't sound like a good place to start the proto migration in the discovery service.

**Review by jakule — APPROVED** [review]

**Review by justinas — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 3]
```diff
@@ -6536,3 +6536,21 @@ message OktaOptions {
...
+}
+
+// AccessGraphAWSSync is a configuration for AWS Access Graph service poll service.
+message AccessGraphAWSSync {
+  // AccountID is the AWS account ID to sync.
+  string AccountID = 1 [(gogoproto.jsontag) = "account_id,omitempty"];
```
Comment (marcoandredinis) [inline api/proto/teleport/legacy/types/types.proto:6548]: Do we need to ask for the account id?
It usually comes from the credentials available.

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (marcoandredinis) [inline api/proto/teleport/legacy/types/types.proto:6542]:
```suggestion
  repeated AccessGraphAWSSync AWS = 1 [(gogoproto.jsontag) = "aws,omitempty"];
```
It currently looks like this:
```yaml
kind: discovery_config
....
spec:
  aws:
    - types: ["ec2"]
      regions: ["x", "y"]
  access_graph:
    types:
      - account_id: 123456789012
        regions: ["x", "y"]
```

I was expecting to have the following, instead:
```yaml
  access_graph:
    aws:
      - account_id: 123456789012
        regions: ["x", "y"]
```

File: api/proto/teleport/legacy/types/types.proto [hunk 4]
```diff
@@ -6536,3 +6536,21 @@ message OktaOptions {
...
   ];
 }
+
+// AccessGraphSync is a configuration for Access Graph service.
+message AccessGraphSync {
+  repeated AccessGraphAWSSync AWS = 1 [(gogoproto.jsontag) = "types,omitempty"];
+}
+
+// AccessGraphAWSSync is a configuration for AWS Access Graph service poll service.
+message AccessGraphAWSSync {
```
Comment (marcoandredinis) [inline api/proto/teleport/legacy/types/types.proto:6546]: I think we should support integrations, it's more future proof

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (tigrato) [reply api/proto/teleport/legacy/types/types.proto:6542]: typo :cry:

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**